Document credentials over non-TLS connections (#3916)
grafana/agent: docs/sources/flow/reference/components/otelcol.exporter.otlp.md

{{< docs/shared lookup="flow/reference/components/otelcol-tls-config-block.md" source="agent" >}}

> > **NOTE**: `otelcol.exporter.otlp` uses gRPC, which disallows sending sensitive credentials (like `auth`) over insecure channels for security reasons.

Reviewer: ```suggestion
> **NOTE**: `otelcol.exporter.otlp` uses gRPC, which does not allow you to send sensitive credentials (like `auth`) over insecure channels.
```

> > If you need to send sensitive credentials over non-TLS connections (via the `insecure` argument), you may want to use [otelcol.exporter.otlphttp][] instead.

Reviewer: ```suggestion
> If you need to send sensitive credentials over non-TLS connections (via the `insecure` argument), you can use [otelcol.exporter.otlphttp][] instead.
```

Reviewer 2: ```suggestion
> Sending sensitive credentials over insecure non-TLS connections is supported by non-gRPC exporters such as [otelcol.exporter.otlphttp][.
```
What about this wording?

Reviewer: From a language standpoint, that's good. It's just a clear statement without giving specific direction (which seems to be more appropriate here).

Author: I am fine with this, @rfratto WDYT?

Reviewer 3: SGTM 👍
